e2e/LongHaul/service/IotHub.cs

```diff
@@ -20,29 +19,24 @@ internal class IotHub : IDisposable
         private readonly Logger _logger;
         private readonly string _hubConnectionString;
         private readonly IotHubTransportProtocol _transportProtocol;
-        private readonly string _deviceId;
         private readonly string _storageConnectionString;
 
-        private static readonly TimeSpan s_directMethodInvokeInterval = TimeSpan.FromSeconds(3);
-        private static readonly TimeSpan s_desiredPropertiesSetInterval = TimeSpan.FromSeconds(3);
-        private static readonly TimeSpan s_c2dMessagesSentInterval = TimeSpan.FromSeconds(3);
         private static readonly TimeSpan s_deviceCountMonitorInterval = TimeSpan.FromSeconds(30);
+        private static readonly TimeSpan s_receiveFileUploadInterval = TimeSpan.FromSeconds(30);
         private int _totalFileUploadNotificationsReceived;
 
         private static IotHubServiceClient s_serviceClient;
-        private static HashSet<string> s_activeDeviceSet;
         private BlobContainerClient _blobContainerClient;
 
-        private long _totalMethodCallsCount = 0;
-        private long _totalDesiredPropertiesUpdatesCount = 0;
-        private long _totalC2dMessagesSentCount = 0;
+        private static volatile Dictionary<string, Action> s_onlineDeviceOperations;
+        private static volatile Dictionary<string, CancellationTokenSource> s_onlineDeviceCancellationTokenSources;
```

Comment:
Consider caching both CTS and the task together in a tuple or a custom class. Then you don't need two dictionaries and risk them getting out of sync from each other.